source/common/config/grpc_subscription_impl.h

```diff
@@ -60,5 +62,20 @@ class GrpcSubscriptionImpl : public Subscription,
 using GrpcSubscriptionImplPtr = std::unique_ptr<GrpcSubscriptionImpl>;
 using GrpcSubscriptionImplSharedPtr = std::shared_ptr<GrpcSubscriptionImpl>;
 
+class GrpcCollectionSubscriptionImpl : public GrpcSubscriptionImpl {
+public:
+  GrpcCollectionSubscriptionImpl(const xds::core::v3::ResourceLocator& collection_locator,
+                                 GrpcMuxSharedPtr grpc_mux, SubscriptionCallbacks& callbacks,
+                                 OpaqueResourceDecoder& resource_decoder, SubscriptionStats stats,
+                                 absl::string_view type_url, Event::Dispatcher& dispatcher,
+                                 std::chrono::milliseconds init_fetch_timeout, bool is_aggregated);
+
+  void start(const std::set<std::string>& resource_names,
```

Comment:
perf nit: switch to flat_hash_set. Follow up is fine if that is easier.

Reply:
I'll do this in a separate PR, since it's an existing virtual method, added to TODO.